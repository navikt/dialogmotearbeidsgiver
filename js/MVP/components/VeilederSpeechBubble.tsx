import React, { ReactElement } from 'react';
import styled from 'styled-components';
import Veileder from 'nav-frontend-veileder';
import VeilederAvatar from '../components/svg/VeilederAvatar';

const VeilederStyled = styled(Veileder)`
  max-width: 576px;
  align-self: center;
  margin: 64px 0;
`;

interface Props {
  content: ReactElement;
}

function VeilederSpeechBubble({ content }: Props): ReactElement {
  return (
    <VeilederStyled tekst={content} posisjon="hÃ¸yre" storrelse="S" fargetema="info" hvitSnakkeboble>
      <VeilederAvatar />
    </VeilederStyled>
  );
}

export default VeilederSpeechBubble;
