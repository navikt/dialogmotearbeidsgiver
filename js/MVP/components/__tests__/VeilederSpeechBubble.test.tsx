import { render } from '@testing-library/react';
import React from 'react';
import VeilederSpeechBubble from '../VeilederSpeechBubble';

describe('VeilederSpeechBubble', () => {
  test('should render', async () => {
    const Content = <div>Test text</div>;
    const { container } = render(<VeilederSpeechBubble content={Content} />);

    expect(container).toHaveTextContent('Test text');
  });
});
