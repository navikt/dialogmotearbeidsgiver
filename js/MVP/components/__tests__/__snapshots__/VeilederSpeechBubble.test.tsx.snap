// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VeilederSpeechBubble should render 1`] = `
<div>
  <div
    class="nav-veileder sc-bdvvtL jQhzHo nav-veileder--hoyre"
  >
    <div
      class="nav-veileder__frame nav-veileder__frame--s nav-veileder__frame--info"
    >
      <img
        src="[object Object]"
      />
    </div>
    <i
      class="nav-veileder__snakkeboblePil nav-veileder__snakkeboblePil--hoyre nav-veileder__snakkeboblePil--hvit"
    />
    <div
      class="nav-veileder__snakkeboble nav-veileder__snakkeboble--hoyre nav-veileder__snakkeboble--hvit"
    >
      <div>
        Test text
      </div>
    </div>
  </div>
</div>
`;
